locking/locks.go

```diff
@@ -127,39 +135,55 @@ type Lock struct {
 	Name string
 	// Email address of the person holding this lock
 	Email string
-	// CommitSHA is the commit that this Lock was created against. It is
-	// strictly equal to the SHA of the minimum commit negotiated in order
-	// to create this lock.
-	CommitSHA string
-	// LockedAt is a required parameter that represents the instant in time
-	// that this lock was created. For most server implementations, this
-	// should be set to the instant at which the lock was initially
-	// received.
+	// LockedAt tells you when this lock was acquired.
 	LockedAt time.Time
-	// ExpiresAt is an optional parameter that represents the instant in
-	// time that the lock stopped being active. If the lock is still active,
-	// the server can either a) not send this field, or b) send the
-	// zero-value of time.Time.
-	UnlockedAt time.Time
 }
 
 func (c *Client) newLockFromApi(a api.Lock) Lock {
 	return Lock{
-		Id:         a.Id,
-		Path:       a.Path,
-		Name:       a.Committer.Name,
-		Email:      a.Committer.Email,
-		CommitSHA:  a.CommitSHA,
-		LockedAt:   a.LockedAt,
-		UnlockedAt: a.UnlockedAt,
+		Id:       a.Id,
+		Path:     a.Path,
+		Name:     a.Committer.Name,
+		Email:    a.Committer.Email,
+		LockedAt: a.LockedAt,
 	}
 }
 
 // SearchLocks returns a channel of locks which match the given name/value filter
 // If limit > 0 then search stops at that number of locks
-func (c *Client) SearchLocks(filter map[string]string, limit int) (locks []Lock, err error) {
+// If localOnly = true, don't query the server & report only own local locks
+func (c *Client) SearchLocks(filter map[string]string, limit int, localOnly bool) (locks []Lock, err error) {
+
+	if localOnly {
+		return c.searchCachedLocks(filter, limit)
+	} else {
+		return c.searchRemoteLocks(filter, limit)
+	}
+}
+
+func (c *Client) searchCachedLocks(filter map[string]string, limit int) ([]Lock, error) {
+	cachedlocks := c.cachedLocks()
+	path, filterByPath := filter["path"]
+	id, filterById := filter["id"]
+	lockCount := 0
+	locks := make([]Lock, 0, len(cachedlocks))
+	for _, l := range cachedlocks {
+		// Manually filter by Path/Id
+		if (filterByPath && path != l.Path) ||
```

Comment:
I follow this logic, but I think we can make it clearer:

```go
for _, l := range cachedLocks {
        matchesPath := path == l.Path
        matchesId := id == l.Id

        if (filterByPath && matchesPath) || (filterById && matchesId) {
                // ...
        }
}
```

This does add a level of indentation to the body of the loop, but I think the extra clarity is worth it.